core/src/main/java/smile/classification/DecisionTree.java

```diff
@@ -577,7 +586,7 @@ public Node findBestSplit(int n, int[] count, int[] falseCount, double impurity,
 
                     int trueLabel = Math.whichMax(trueCount[l]);
                     int falseLabel = Math.whichMax(falseCount);
-                    double gain = impurity - (double) tc / n * impurity(trueCount[l], tc) - (double) fc / n * impurity(falseCount, fc);
+					double gain = impurity - (double) tc / n * impurity(trueCount[l], tc) - (double) fc / n * impurity(falseCount, fc);
```

Comment:
With missing values, can we really still use this formula?

Reply:
The question I also had in mind. I haven't found the formula in literature... Maybe you can give me a hint on where I can read about it?

Reply:
@haifengl any idea where I can find out about this formula?

Reply:
Any book or papers on decision tree or CART should have it.